dLux/apertures.py

```diff
@@ -0,0 +1,1430 @@
+            The angle away from the positive x-axis of the coordinate
+            system associated with this aperture. Please note that `theta`
+            May not start at zero. 
+
+        Returns 
+        -------
+        wedges : Tensor 
+            The angular bounds associated with each pair of vertices in 
+            order. The leading axis of the Tensor steps through the 
+            wedges in order arround the circle. 
+        """
+        return (self.alpha[:-1] < theta) & (theta < self.alpha[1:])
+
+
+    def _segments(self : Layer, theta : Tensor, rho : Tensor) -> Tensor:
+        """
+        Generate the segments as a stacked tensor. 
+
+        Parameters
+        ----------
+        theta : Tensor
+            The angle of every pixel associated with the coordinate system 
+            of this aperture. 
+        rho : Tensor
+            The radial positions associated with the coordinate system 
+            of this aperture. 
+
+        Returns 
+        -------
+        segments : Tensor 
+            The bitmaps corresponding to each vertex pair in the vertices.
+            The leading dimension contains the unique segments. 
+        """
+        edges = self._edges(rho, theta)
+        wedges = self._wedges(theta)
+        return (edges & wedges).astype(float)
+        
+
+    def _aperture(self : Layer) -> Matrix:
+        """
+        Generate the BitMap representing the aperture described by the 
+        vertices. 
+
+        Returns
+        -------
+        aperture : Matrix 
+            The Bit-Map that represents the aperture. 
+        """
+        rho, theta = self._rad_coordinates()
+        segments = self._segments(theta, rho)
+        return segments.sum(axis=0)
+
+
+class CompoundAperture(eqx.Module):
+    """
+    Represents an aperture that contains more than one single 
+    aperture. The smaller sub-apertures are stored in a dictionary
+    pytree and are so acessible by user defined name. For example:
+
+    >>> x_sep = 0.1
+    >>> width = 0.005
+    >>> height = 0.2
+    >>> first_slit = RectangularAperture(
+    ...     npix=1008, width=width, height=height, 
+    ...     x_offset = -x_sep/2, y_offset=0.,
+    ...     theta=0., phi=0., magnification=0.)
+    >>> second_slit = RectangularAperture(
+    ...     npix=1008, width=width, height=height, 
+    ...     x_offset = x_sep/2, y_offset=0.,
+    ...     theta=0., phi=0., magnification=0.)
+    >>> apertures = {"Right": first_slit, "Left": second_slit}
+    >>> double_slit = CompoundAperture(apertures)
+    >>> double_slit["Right"]
+
+    Attributes
+    ----------
+    npix : int
+        The number of pixels to be used to represent the entire 
+        set of apertures.
+    pixel_scale : float, meters per pixel
+        The length of the side of one pixel in the image. 
+    apertures : dict(str, Layer)
+        The apertures that make up the compound aperture. 
+    """
+    npix : int
+    pixel_scale : float
```

Comment:
Same as above, we don't want multiple pixel scales, this should be taken from the wavefront and applied in the __call__ method

Reply:
Yeah, I'll switch to a `width_of_image` parameter like in the `Spider` class. I think the `Spider` class may also extend aperture in the future so that it can be used more easily with `CompoundAperture`.